Sources/Core/Src/my_app/app_inject_simulator/inject_simulator_api.h

```diff
@@ -0,0 +1,74 @@
+/**
+ * @file inject_simulator.h
+ * @author Nicolás Almaraz
+ * @brief Current injection simulator
+ *
+ * This API is responsible for simulating what would be a current injection into the power line.
+ * Each time it receives the notification that a new cycle has started, it injects the pre-loaded waveform via DAC.
+ * This allows the waveform calculated by the microcontroller to be measured with an oscilloscope at the DAC output.
+ */
+
+#ifndef SRC_MY_APP_INJECT_SIMULATOR_INJECT_SIMULATOR_H_
+#define SRC_MY_APP_INJECT_SIMULATOR_INJECT_SIMULATOR_H_
+
+//======================================
+// Dependencies
+//======================================
+#include <stdbool.h>
+#include <stdint.h>
+
+#include "my_app.h"
+
+//======================================
+// Public Defines
+//======================================
+
+//======================================
+// Public Structures and Data Types
+//======================================
+
+//======================================
+// Global Variables
+//======================================
+
+//======================================
+// Public Function Declarations
+//======================================
+
+/**
+ * @brief Initializes the API
+ */
+void inject_simulator_api_init();
+
+/**
+ * @brief Manages the start of a new transmission.
+ * This function should run in the main loop
+ */
+void app_inject_simulator_loop();
+
+/**
+ * @brief Sets the current waveform to inject on each new cycle detection
+ * @param buffer with the cycle to inject
+ */
+void inject_simulator_api_set_current_waveform(cycle_t cycle);
+
+/**
+ * @brief Enable or disable the injection
+ * @param enable true: Enable -- false: Disable
+ */
+void inject_simulator_api_set_enable(bool enable);
+
+/**
+ * @brief Notifies the API that a new cycle has started
+ * This function should be called whenever a new cycle is detected on the power line
+ */
+void inject_simulator_api_cycle_callback();
```

Comment:
Yo leo "callback" y me creo que es algo del usuario.
Es esto un callback ? Realmente este módulo le pasa esa función a otro ? Nope, lo que veo es que otro módulo llama a esta función, con lo que, para mí, esto no es un callback. There is no back call, no calling back, hence it aint't no freaking callback !

... si fuera un callback, no tendría por qué estar en el header, (por eso asumo inmediatamente que es algo que se le exige al usuario (otra mala práctica) al estilo FreeRTOS+TCP, para indicarle eventos), ya que, quien llama, se la pasa, y sería una estática.

Reply:
Sí, quedó raro. Acá tenía dudas de cómo ponerle a los nombres de las funciones. 

Lo que quise hacer es hacer una función correspondiente a ese módulo que se llame cuando llega una interrupción. Es algo así como signals y slots. 

La "signal" sería la interrupción del GPIO y el "slot" sería "inject_simulator_api_cycle_callback". 

Esas funciones "slot" que están en unas cuantas APIs se llaman desde el archivo "callback_functions_manager.c". Allí dentro están las funciones callback de todos los eventos e interrupciones del micro.

Reply:
Una "signal" es algo que se genera externo. A lo sumo sería un `signal_handler`, y no es más que una función de la API, porque para eso existe una API: para poder saber cómo llamar a ese módulo bajo esas condiciones.

Reply:
Bien, acá en la nueva revisión le puse "IRQHandler". Esa era la palabra que necesitaba en lugar de callback.